File: my-folder/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        max_len = 0
    
        for num in nums:
            if num - 1 not in nums:
                c = 0
                while(c + num in nums):
                    c += 1
                    max_len = max(max_len, c)
        
        return max_len

        

File: my-folder/0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> store = new HashSet<>();
        for(int num : nums){
            store.add(num);
        }

        int longest = 0;

        for(int num : store){
            int curr = 0;
            if(!store.contains(num - 1)){
                int currNum = num;
                while(store.contains(currNum)){
                    curr += 1;
                    currNum += 1;
                }

                longest = Math.max(longest, curr);
            }
        }

        return longest;
    }
}
